Sources/SwiftMemcache/MemcachedConnection.swift

```diff
@@ -0,0 +1,198 @@
+///
+/// This actor can be used to send commands to the server.
+public actor MemcachedConnection {
+    private typealias StreamElement = (MemcachedRequest, CheckedContinuation<MemcachedResponse, Error>)
+    private let host: String
+    private let port: Int
+
+    /// Enum representing the current state of the MemcachedConnection.
+    ///
+    /// The State is either initial, running or finished, depending on whether the connection
+    /// to the server is active or has been closed. When running, it contains the properties
+    /// for the buffer allocator, request stream, and the stream's continuation.
+    private enum State {
+        case initial(
+            /// The channel's event loop group.
+            eventLoopGroup: EventLoopGroup,
+            /// The allocator used to create new buffers.
+            bufferAllocator: ByteBufferAllocator,
+            /// The stream of requests to be sent to the server.
+            requestStream: AsyncStream<StreamElement>,
+            /// The continuation for the request stream.
+            requestContinuation: AsyncStream<StreamElement>.Continuation
+        )
+        case running(
+            /// The allocator used to create new buffers.
+            bufferAllocator: ByteBufferAllocator,
+            /// The underlying channel to communicate with the server.
+            channel: NIOAsyncChannel<MemcachedResponse, MemcachedRequest>,
+            /// The stream of requests to be sent to the server.
+            requestStream: AsyncStream<StreamElement>,
+            /// The continuation for the request stream.
+            requestContinuation: AsyncStream<StreamElement>.Continuation
+        )
+        case finished
+    }
+
+    /// Enum representing the possible errors that can be encountered in `MemcachedConnection`.
+    enum MemcachedConnectionError: Error {
+        /// Indicates that the connection has shut down.
+        case connectionShutdown
+    }
+
+    private var connectionState: State
+
+    /// Initialize a new MemcachedConnection.
+    ///
+    /// - Parameters:
+    ///   - host: The host address of the Memcache server.
+    ///   - port: The port number of the Memcache server.
+    ///   - eventLoopGroup: The event loop group to use for this connection.
+    public init(host: String, port: Int, eventLoopGroup: EventLoopGroup) {
+        self.host = host
+        self.port = port
+        let (stream, continuation) = AsyncStream<StreamElement>.makeStream()
+        let bufferAllocator = ByteBufferAllocator()
+        self.connectionState = .initial(
+            eventLoopGroup: eventLoopGroup,
+            bufferAllocator: bufferAllocator,
+            requestStream: stream,
+            requestContinuation: continuation
+        )
+    }
+
+    /// Start consuming the requestStream
+    ///
+    /// This function starts consuming the requests from the `requestStream`,
+    /// sending each request to the Memcache server and handling the server's responses.
```

Comment:
This is talking more about the internals than the public API
```suggestion
    /// Runs the Memcache connection.
    ///
    /// This method connects to the Memcache server and starts handling requests. It only returns when the connection
    /// to the server is finished or the task that called this method is cancelled. 
```